### ISSUE TITLE
PyGnome and WebGnomeAPI

### ISSUE THREAD

**Issue by Yoavp81** [NOAA-ORR-ERD/PyGnome#124]
Hi,

I'm fairly new to python, Conda and scripting. Since WebGnome hangs when running it, I would like to run PyGnome via WebGnomeAPI.
What is the correct order of installation of all the components and dependencies necessary?
I've installed PyGnome and oil dependencies and then switched to installing WebGnomeAPI but was unable to run the server.

Thank you,
Yoav

**Comment by jay-hennen** [comment]
Just so we know, what kind of hangups are you getting on WebGNOME? It's an important project for us.

To be honest if you are planning to not use the client, it would be far better to skip the API entirely and use PyGNOME directly. The API is only there to facilitate web services and client communication.

**Comment by Yoavp81** [comment]
Thank you for the response.
I've created a model and initially it for a certain amount of timesteps, producing output, but beyond a certain point, it hangs, no matter how long I keep it running (with no errors or warnings). I've tried decreasing the model's resolution and changed it to a daily timestep. That seems to help the model to finalize but I did not receive any spatial dispersion.

I'll try using PyGNOME but was wondering if there is any tool that helps convert the .gnome file or inputs into PyGNOME Python script? that would be very helpful.

Thanks again

**Comment by jay-hennen** [comment]
The .gnome files are save files and are compatible with PyGNOME.

```
from gnome.model import Model
m = Model.load('savefile.gnome')
m.full_run()
```
In theory that's enough to do a model run. If you are using a savefile created on webGNOME you would want to add an outputter, such as NetCDFOutput or Renderer (see included scripts in pygnome for more details `/py_gnome/scripts/example_scripts/`)

If it's not too large, could you share your savefile with us? I want to investigate the cause of the hangups.

**Comment by Yoavp81** [comment]
It seems that using the savefile in PyGNOME also doesn't work.
I'm attaching it here.
Thank you
[Model.zip](https://github.com/NOAA-ORR-ERD/PyGnome/files/9007336/Model.zip)

**Comment by coconnor8** [comment]
I think the issue is the 235000 particles. I'm not sure if that should work
and is a bug, or if
we need to put in a limit. For an instantaneous spill you should be able to
run with a lot
fewer particles. It worked fine for an order of magnitude or two fewer.

The error message was - WebSocket connection to 'wss://
gnome-dev.orr.noaa.gov/socket.io/?EIO=4&transport=websocket&sid=om9iBnBYY73Qk-DaAAR8'
failed: One or more reserved bits are on: reserved1 = 1, reserved2 = 1,
reserved3 = 1

**Comment by Yoavp81** [comment]
Yes, that was the issue. It runs smoothly now on WebGNOME and PyGNOME.
Thank you!

**Comment by jay-hennen** [comment]
235,000 elements is far too many! Considering computational expense scales linearly with the number of elements I'm happy to see it ran at all.

Consider that a "normal" model run would use only 1000 elements!

**Comment by AmyMacFadyen** [comment]
Just a quick note -- if you do have a need to use a lot of particles (e.g. statistical analysis) you could consider  splitting into multiple runs and combining the results.

For WebGNOME there probably should be a hard limit.
